database/prod/appointmentDB.cpp

```diff
@@ -0,0 +1,113 @@
+#include "appointmentDB.h"
+
+#include <QString>
+
+const QString ID_FIELD = "id";
+
+AppointmentDB::AppointmentDB(
+    HomeopathyDB* homeopathy,
+    MedicineDB* medicine,
+    ServiceDB* service,
+    PatientDB* patient)
+{
+    tableName_ = "appointments";
+    homeopathyDb_ = homeopathy;
+    medicineDb_ = medicine;
+    serviceDb_ = service;
+    patientDb_ = patient;
+
+    columns_ = {
+        "id", "isDeleted", "date", "isConducted",
+        "homeopathy", "service", "patient"
+    };
+}
+
+void AppointmentDB::handleListResult(QSqlQuery& query, std::vector<Appointment>& receiver) {
+    Appointment current;
+
+    while (query.next()) {
+        Appointment next =rowToItem(query);
+
+        if (next == current) {
+            current.medicines.push_back(next.medicines[0]);
+            continue;
+        }
+
+        receiver.push_back(current);
+        current = next;
+    }
+}
+
+Appointment AppointmentDB::rowToItem(QSqlQuery row, int startPos) {
+    int columnsShift = columns_.size();
+
+    Appointment result;
+
+    if (row.value(0 + startPos).toInt() == 0) {
+        return result;
+    }
+
+    setCode(result, row.value(0 + startPos).toInt());
+    setDeleted(result, row.value(1 + startPos).toBool());
+    result.date = row.value(2 + startPos).toDateTime();
```

Comment:
```suggestion
    setDeleted(result, row.value(1 + startPos).toBool());
    
    result.date = row.value(2 + startPos).toDateTime();
```

// в других базах этот отступ есть, добавляю для единообразия